### PR TITLE
FIX Errors in requirements and message collector in CMS, and badge alignment

### PR DESCRIPTION
**Pull request by robbieaverill**
This fixes a couple of errors when hitting `/admin/pages` since the Requirements API changed and we implemented a Monolog bridge instead of a collector for messages. It also fixes alignment for the number badges on the debug bar tabs.

### CODE REVIEW

**Comment by robbieaverill** [comment]
cc @Firesphere

File: assets/debugbar.css [hunk 1]
```diff
@@ -157,6 +157,7 @@ a.phpdebugbar-tab.phpdebugbar-active span.phpdebugbar-badge {
   a.phpdebugbar-tab span.phpdebugbar-badge {
     display: none;
     margin-left: 5px;
+    float: right;
```
Comment (Firesphere) [inline assets/debugbar.css:160]: I don't think a PR should address 2 issues at the same time. I've copied this minor change to my `Use Laravel Mix` PR.

A PR should cover a single issue, not multiple.

File: code/DebugBar.php [hunk 2]
```diff
@@ -201,7 +201,7 @@ public static function includeRequirements()
         }
         // If jQuery is already included, set to false
         $js = Requirements::backend()->getJavascript();
-        foreach ($js as $url) {
+        foreach ($js as $url => $args) {
             $name = basename($url);
```
Comment (Firesphere) [inline code/DebugBar.php:205]: Won't this break if it's a numeric assoc array? e.g. `[0 => 'myfile.css', 1 => 'myfile.js']`

I believe a sanity check e.g.
```
if (is_array($args)) {
    $name = $url;
} else {
   $name = $args;
}
```
would make more sens

**Review by Firesphere — CHANGES_REQUESTED** [review]

File: code/DebugBar.php [hunk 3]
```diff
@@ -201,7 +201,7 @@ public static function includeRequirements()
...
             if ($name == 'jquery.js' || $name == 'jquery.min.js') {
```
Comment (Firesphere) [inline code/DebugBar.php:206]: Although this is existing code, I'm not really sure it's a good idea to just check these two. A minified version of jquery might be included in a packed JS, including a new version of jquery will break all that.

A better approach would be to see if jquery exists in the javascript, and if not, add it dynamically.

**Review by Firesphere — CHANGES_REQUESTED** [review]

File: code/DebugBar.php (hunk 3 above)
Comment (robbieaverill) [reply code/DebugBar.php:206]: Yeah, fair comment. Feel free to open an issue or a pull request to address this.

**Review by fspringveldt — APPROVED** [review]

File: code/DebugBar.php (hunk 2 above)
Comment (robbieaverill) [reply code/DebugBar.php:205]: Requirements provide the path as the key and an array of arguments as the values in SS4. See [here](https://github.com/silverstripe/silverstripe-framework/blob/4/src/View/Requirements_Backend.php#L432-L436).

File: assets/debugbar.css (hunk 1 above)
Comment (robbieaverill) [reply assets/debugbar.css:160]: I agree in principle, but separating these changes into two commits or PRs would be over the top. This PR is just to get the CMS working again for SS4.

**Pull request merged**